internal/handlers/common/findandmodify.go

```diff
@@ -96,21 +97,24 @@ func PrepareFindAndModifyParams(document *types.Document) (*FindAndModifyParams,
 			update = updateParam
 		case *types.Array:
 			// TODO aggregation pipeline stages metrics
-			return nil, NewErrorMsg(ErrNotImplemented, "Aggregation pipelines are not supported yet")
+			return nil, NewCommandErrorMsgWithArgument(ErrNotImplemented, "Aggregation pipelines are not supported yet", "update")
 		default:
-			return nil, NewErrorMsg(ErrFailedToParse, "Update argument must be either an object or an array")
+			return nil, NewCommandErrorMsgWithArgument(ErrFailedToParse, "Update argument must be either an object or an array", "update")
 		}
 	}
 
 	if update != nil && remove {
+		// TODO: update or remove?
 		return nil, NewErrorMsg(ErrFailedToParse, "Cannot specify both an update and remove=true")
```

Comment:
There are couple of places like this. It's hard to specify which parameter should be used as an argument

Reply:
Neither, the current code is fine